stdlib/public/core/Set.swift

```diff
@@ -1223,7 +1223,11 @@ extension Set {
   @inlinable
   public __consuming func intersection(_ other: Set<Element>) -> Set<Element> {
     var newSet = Set<Element>()
-    for member in self {
+    var (iter, other) = (self, other)
```

Comment:
I think another option instead of swapping could be have the implementing on an internal function 
```swift
@inline(__always)
internal func __intersection(_ other: Set<Element>) -> Set<Element> {
     var newSet = Set<Element>()     
     for member in iter {
       if other.contains(member) {
         newSet.insert(member)
       }
    }
    return newSet
  }
```
And then 
```swift
@inlinable
public __consuming func intersection(_ other: Set<Element>) -> Set<Element> {
  return count > other.count 
         ?  other.__intersection(self) 
         :  self.__intersection(other)
}
```
But is only a suggestion :)

Reply:
What happens if you just write `guard count <= other.count else { return other.intersection(self) }`?

Reply:
I think we should go with whatever results in the smallest code size! Swapping seems like a nice trick that achieves that. In similar situations I sometimes [used recursion](https://github.com/apple/swift/pull/21300/files#diff-5c7a8449f14e168ec3e7aa580cecc46e3cf78965229ad530f0853982e977a149R676-R691), but I wonder if swapping simplifies optimizations.

(Using a conditional expression could maybe lead to the implementation getting inlined twice. Or maybe it doesn't -- it depends on the whims of the optimizer though, and if we have the luxury of choosing between multiple alternatives, we should probably go with one that doesn't expect too much of the compiler. 😄)

Reply:
Fair enough! Thanks @lorentey :)